src/com/intellij/idea/plugin/hybris/impex/assistance/ImpexColumnBreadcrumbsProvider.kt

```diff
@@ -0,0 +1,94 @@
+package com.intellij.idea.plugin.hybris.impex.assistance;
+
+import com.intellij.idea.plugin.hybris.impex.ImpexLanguage
+import com.intellij.idea.plugin.hybris.impex.psi.*
+import com.intellij.idea.plugin.hybris.impex.utils.ImpexPsiUtils
+import com.intellij.lang.Language
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.ui.breadcrumbs.BreadcrumbsProvider
+import com.intellij.util.ObjectUtils
+
+class ImpexColumnBreadcrumbsProvider : BreadcrumbsProvider {
+
+    override fun getLanguages(): Array<Language> = arrayOf(ImpexLanguage.getInstance())
+
+    override fun acceptElement(element: PsiElement) = element is ImpexFullHeaderParameter
+        || element is ImpexFullHeaderType
+        || element is ImpexAnyHeaderMode
+
+    override fun getElementInfo(psi: PsiElement): String {
+        val headerParameter = getLinkedHeaderParameter(psi)
+        if (headerParameter != null) return headerParameter.text
+
+        val adjustedPsi = adjustWhiteSpaceAndSeparator(psi)
+
+        val parentParameter = PsiTreeUtil.getParentOfType(adjustedPsi, ImpexFullHeaderParameter::class.java, false)
```

Comment:
There is method get parent if any type, or smth like that.
Refactor in a way we will find parent and then will use when statement with exact class-to-logic mapping